src/mmw/js/src/core/modals/views.js

```diff
@@ -152,10 +152,10 @@ var ShareView = ModalBaseView.extend({
 
             confirm.render();
             confirm.on('confirmation', function() {
-                var project = self.options.app.currentProject;
+                var project = self.options.project || self.options.app.currentProject;
 
                 project.set('is_private', false);
-                project.saveProjectAndScenarios();
```

Comment:
If you share your project immediately after creating it now will it save the scenarios?